vaccine-standby/functions/get-studio-executions.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-console */
+const crypto = require('crypto');
+
+// eslint-disable-next-line func-names
+exports.handler = function (context, event, callback) {
+  // Validate token before running
+  function isAllowed(token) {
+    // Create the token with the environment password
+    const tokenString = `${context.ACCOUNT_SID}:${context.ADMIN_PASSWORD}:${context.SALT}`;
+
+    // Similar to TwilioClient
+    const originalToken = crypto
+      .createHmac('sha1', context.AUTH_TOKEN)
+      .update(Buffer.from(tokenString, 'utf-8'))
+      .digest('base64');
+
+    return originalToken === token;
+  }
+
+  if (!isAllowed(event.token)) {
+    // eslint-disable-next-line no-undef
+    const response = new Twilio.Response();
+    response.status(403);
+    response.message('unathorized');
+
+    callback('unauthorized', response);
+    return;
+  }
+
+  const client = context.getTwilioClient();
+
+  client.studio.v1.flows(event.sid).executions.list({ limit: 100 })
+    .then((executions) => {
+      if (executions.length > 0) {
+        const endedExecutions = executions.filter((e) => e.status === 'ended');
+        const promises = endedExecutions.map((e) => new Promise((resolve, reject) => {
```

Comment:
I don't think you need this wrapped in a promise, if you change this to `return client.studio.flows` and wherever you use `resolve` you can just `return` on the data. In the `reject` case it would probably just be `callback(err)`

Reply:
I tried this but it did not work. I do think the promise wrapper is needed here.